source/server/listener_manager_impl.h

```diff
@@ -100,9 +105,22 @@ class ProdListenerComponentFactory : public ListenerComponentFactory,
   createDrainManager(envoy::config::listener::v3::Listener::DrainType drain_type) override;
   uint64_t nextListenerTag() override { return next_listener_tag_++; }
 
+  Filter::FilterConfigProviderPtr<Network::ListenerFilterFactoryCb>
+  createStaticFilterConfigProvider(const Network::ListenerFilterFactoryCb& callback,
+                                   const std::string& filter_config_name) override {
+    return tcp_listener_config_provider_manager_->createStaticFilterConfigProvider(
+        callback, filter_config_name);
+  }
+
+  Filter::TcpListenerFilterConfigProviderManagerImpl& getTcpListenerConfigProviderManager() {
+    return *tcp_listener_config_provider_manager_;
+  }
+
 private:
   Instance& server_;
   uint64_t next_listener_tag_{1};
+  std::unique_ptr<Filter::TcpListenerFilterConfigProviderManagerImpl>
```

Comment:
OOC why is this a unique pointer, and not a plain member?

Reply:
Yes, this can be a plain member. Will make the change.